explainaboard/metrics/metric.py

```diff
@@ -371,16 +371,41 @@ def get_batch_data(self) -> np.ndarray[tuple[int, int, int], Any]:
 
 
 class Metric(metaclass=abc.ABCMeta):
-    """A class representing an evaluation metric."""
+    """A class representing an evaluation metric.
+
+    When a subclass needs to construct a random number generator, initialize the random
+    generator as follows:
+
+    1. Invoke the `get_seed()` method to get a numpy SeedSequence;
+    2. Spawn the SeedSequence with the SeedSequence's `spawn` method;
+    3. Pass the spawned SeedSequence to the random generator's constructor.
+
+    Example:
+        class FooMetric(Metric):
+            def foo(self) -> None:
+                # Spawns a SeedSequence.
+                seed = self.get_seed().spawn(1)[0]
+                # Initializes a random generator with the spawned seed.
+                rng = np.random.default_rng(seed)
+                # Do something with `rng`.
+
+        config = MetricConfig(...)
+        metric = FooMetric(config, seed=np.random.SeedSequence(12345))
+        metric.foo()
+    """
 
     config: MetricConfig
 
-    def __init__(self, config: MetricConfig):
+    def __init__(self, config: MetricConfig, seed: np.random.SeedSequence | None):
```

Comment:
It looks the default value `None` has been removed by the new commit. Is there any intent of this?

Reply:
Ah, I was confused with the previous sample code which doesn't have None as the default value. Will bring back the default value of None.

Reply:
Reverted the change which removed `None`.